### ISSUE TITLE
parse_stream gives incompatible type "IO[bytes]"; expected "BinaryIO"

### ISSUE THREAD

**Issue by shearer12345** [timrid/construct-typing#19]
I'm parsing a stream from a zip file, and mypy shows the following error
`Argument 1 to "parse_stream" of "Construct" has incompatible type "IO[bytes]"; expected "BinaryIO"  [arg-type] (11:32)`

I believe that `parse_stream` should also be happy with `IO[bytes]` - see https://github.com/python/typeshed/issues/7843 for some discussion - but I'm a little out of my depth on this part of Python typing.

The following examples will reproduce this mypy error

```
from zipfile import ZipFile

from construct import Struct, GreedyBytes

s = Struct(
    'test' / GreedyBytes
)

with ZipFile('spam.zip') as myzip:
    with myzip.open('eggs.txt') as myfile:
        parsed = s.parse_stream(myfile)
```

Changing `StreamType = t.BinaryIO` to `StreamType = t.Union[t.BinaryIO, t.IO[bytes]]` in construct-stubs/core.pyi resolves this.
I'm happy to create a PR if that seems like the appropriate solution

Thanks so much for construct-typing :)

**Comment by timrid** [comment]
Thanks for the report and sorry for the late reply. I fixed this in [v0.5.6](https://github.com/timrid/construct-typing/releases/tag/v0.5.6).

**Comment by shearer12345** [comment]
Great. Thanks @timrid
